AnnoDesigner/Undo/Operations/ModifyObjectPropertiesOperation.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace AnnoDesigner.Undo.Operations
+{
+    public class ModifyObjectPropertiesOperation<TObject, TProperty> : BaseOperation
+    {
+        private string propertyName;
+        private PropertyInfo propertyInfo;
+
+        public string PropertyName
+        {
+            get { return propertyName; }
+            set
+            {
+                propertyName = value;
+                propertyInfo = typeof(TObject).GetProperty(value);
+                if (!propertyInfo.PropertyType.IsAssignableFrom(typeof(TProperty)))
```

Comment:
I wonder if it would make a big enough perf difference to only do this sanity check in Debug mode?

Reply:
Performance difference will be really minimal, since it is only done once per undoable action like moving of buildings. Also the check itself should be fast. But it definitely will be faster to not have this in Release mode.

Reply:
If its only done once per undoable action, then that should be fine.